python/cudf/cudf/core/index.py

```diff
@@ -776,6 +765,68 @@ def _split(self, splits):
     def _binaryop(self, other, op: str):
         return self._as_int64()._binaryop(other, op=op)
 
+    @property  # type: ignore
+    @_cudf_nvtx_annotate
+    def _column(self):
+        return self._as_int64()._column
+
+    @property  # type: ignore
+    @_cudf_nvtx_annotate
+    def _columns(self):
+        return self._as_int64()._columns
+
+    @property  # type: ignore
+    @_cudf_nvtx_annotate
+    def values_host(self):
+        return self.to_pandas().values
+
+    @_cudf_nvtx_annotate
+    def argsort(
+        self,
+        ascending=True,
+        na_position="last",
+    ):
+        if na_position not in {"first", "last"}:
+            raise ValueError(f"invalid na_position: {na_position}")
+
+        indices = cupy.arange(0, len(self))
+        if (ascending and self._step < 0) or (
+            not ascending and self._step > 0
+        ):
+            indices = indices[::-1]
+        return indices
+
+    def where(self, cond, other=None, inplace=False):
+        return self._as_int64().where(cond, other, inplace)
+
+    def to_numpy(self):
+        return self._as_int64().to_numpy()
+
+    def replace(self, to_replace, value):
+        return self._as_int64().replace(to_replace=to_replace, value=value)
+
+    def to_arrow(self):
+        return self._column.to_arrow()
```

Comment:
Looks like this isn't quite there yet. You replaced `_values` with `_column`, but that has the same effect of creating a Column. Remember that the goal of this function is to generate a PyArrow array. It turns out that you can actually generate a PyArrow array directly from a range object, which would completely avoid creating a column. 
```
>>> import pyarrow
>>> idx = cudf.RangeIndex(10)
>>> idx._range
range(0, 10)
>>> pyarrow.array(idx._range)
<pyarrow.lib.Int64Array object at 0x7f5d08d645e0>
[
  0,
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9
]
>>> idx.to_arrow()
<pyarrow.lib.Int64Array object at 0x7f5d08d64520>
[
  0,
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9
]
```